### PR TITLE
Return position with attributes from S2Meta

### PR DESCRIPTION
**Pull request by catevita**
- [x] return position and attributes meta in one iterator
- [x] prevent accidental modification of s2 meta by making struct members private
- [x] fixing some clippy warnings for AttributeDataType

### CODE REVIEW

**Comment by catevita** [comment]
PTAL @feuerste fyi @nnmm

File: src/lib.rs [hunk 1]
```diff
@@ -105,7 +105,7 @@ impl AttributeDataType {
         Ok(attr)
     }
 
-    pub fn size_of(&self) -> usize {
+    pub fn size_of(self) -> usize {
```
Comment (feuerste) [inline src/lib.rs]: Why consume `self`? We just want to know its size...

Comment (feuerste) [reply src/lib.rs]: Ah, that's because above `Copy` was introduced. What do we need `Copy` for?

Comment (nnmm) [reply src/lib.rs]: I don't think it was needed per se, but for small structs, it's a small convenience and efficiency win with no downsides.

Comment (catevita) [reply src/lib.rs]: Copy was added for the attribute+xyz iterator returned from S2Meta

**Pull request merged**